app/controllers/sites/permissions_controller.rb

```diff
@@ -0,0 +1,60 @@
+module Sites
+	class PermissionsController < ApplicationController
+		include Dcv::RestrictableController
+		include Cul::Omniauth::AuthorizingController
+		before_filter :load_subsite
+		before_filter :authorize_site_update
+		layout 'sites'
+
+		def load_subsite
+			@subsite ||= begin
+				site_slug = params[:site_slug]
+				site_slug = "restricted/#{site_slug}" if restricted?
+				Site.find_by(slug: site_slug)
+			end
+		end
+
+		def update
+			begin
+				update_attributes = permissions_params
+				update_attributes.dig('permissions')&.tap {|atts| @subsite.permissions.assign_attributes(atts) }
+				update_attributes.dig('editor_uids')&.tap {|atts| @subsite.editor_uids = atts }
```

Comment:
Could replace `dig`s with regular hash access?  Also, why use tap here on these lines?

Reply:
the `dig` is an artifact and could be `fetch`, but the `.tap` is for lazy evaluator - don't want to call assign w/ nil

Reply:
Oh!  Good use of tap.  I like that.

For the dig replacement, wouldn't `update_attributes.dig['permissions']` work too, since you'd still get nil if the key isn't present?

Reply:
without the `dig`, yeah - as elsewhere in the controller, actually. Will do.

Reply:
done and updated to use strong params (since this controller does not have complex/deeply nested params)